Apply credential tagging policies at issuer registration

andrewwhitehead opened the pull request.

swcurran commented:
@andrewwhitehead - does self-proving in ICOB still work?  That's going to be the thing to check once the tagging policy is applied.

andrewwhitehead commented:
@swcurran The other PR by @esune addresses most of the issues with the proof request, but will need adjusting to take the tag policy into account and only restrict by the available tags.
